Tests/LibGL/TestShaders.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022, Stephan Unverwerth <s.unverwerth@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/LexicalPath.h>
+#include <AK/String.h>
+#include <LibCore/FileStream.h>
+#include <LibGL/GL/gl.h>
+#include <LibGL/GLContext.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/QOIWriter.h>
+#include <LibTest/TestCase.h>
+
+#ifdef __serenity__
+#    define REFERENCE_IMAGE_DIR "/usr/Tests/LibGL/reference-images"
+#else
+#    define REFERENCE_IMAGE_DIR "reference-images"
+#endif
+#define SAVE_OUTPUT false
+
+static NonnullOwnPtr<GL::GLContext> create_testing_context(int width, int height)
+{
+    auto bitmap = MUST(Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRx8888, { width, height }));
+    auto context = MUST(GL::create_context(*bitmap));
+    GL::make_context_current(context);
+
+    // Assume some defaults for our testing contexts
+    glFrontFace(GL_CCW);
+    glCullFace(GL_BACK);
+    glEnable(GL_CULL_FACE);
+
+    glClearColor(0.0f, 0.0f, 0.0f, 1.0f);
+    glClear(GL_COLOR_BUFFER_BIT);
```

Comment:
These five statements should have been removed from the other test suite; they do nothing except for `glCullFace`, which isn't tested in any of the tests. You can leave them out and just use the context's defaults.

Reply:
Done. I left the lines in place in the TestRender test for now.